src/map/ai/helpers/pathfind.cpp

```diff
@@ -456,9 +449,16 @@ bool CPathFind::FindPath(const position_t& start, const position_t& end)
         return false;
     }
 
-    m_points       = m_POwner->loc.zone->m_navMesh->findPath(start, end);
+    auto [result, points] = m_POwner->loc.zone->m_navMesh->findPath(start, end);
+
+    m_points       = std::move(points);
     m_currentPoint = 0;
 
+    if (result != CNavMesh::PathResult::Complete && wallhack)
+    {
+        m_points.emplace_back(pathpoint_t{ end, 0, false }); // this prevents exploits with navmesh / impassible terrain
```

Comment:
TODO: Add logging here: "I'm going off-mesh after a certain target!", makes for a good indicator of where we have navmesh problems or abuse